### PR TITLE
Add doc

### PR DESCRIPTION
**Pull request by bakura10**
This PR renames all the doc files (sometimes alphabetical order does not work... ;)), and add a new "cookbook" that would be aimed to answer common question and recommended solution.

### CODE REVIEW

**Comment by samsonasik** [comment]
bookdown.json need to be updated

**Comment by weierophinney** [comment]
I'd prefer not to use numbers for ordering; numbers make reordering and having permalinks more difficult. This is why:

- order is as represented in the bookdown.json file
- we include directions for rendering documentation via bookdown
- we will eventually offer rendered docs online (likely quite soon, via a process that will convert bookdown.json to the yaml+markdown format rtfd.org uses)

I definitely want the cookbook, though!

**Comment by bakura10** [comment]
Fixed.

File: doc/book/cookbook.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+```php
+$middleware1 = new UserMiddleware();
+$middleware2 = new ProjectMiddleware();
+
+$app = AppFactory::create();
+$app->pipe($middleware1);
+$app->pipe($middleware2);
+
+$finalApp = AppFactory::create();
+$finalApp->pipe('/api', $finalApp);
```
Comment (ojhaujjwal) [inline doc/book/cookbook.md]: Do you mean `$finalApp->pipe('/api', $app);`?

**Comment by bakura10** [comment]
Yes, fixed!

**Pull request closed without merging**